### PR TITLE
Automate PIP Releases

### LINKED ISSUE #24: New versions should be published to PyPI automatically
We should use GitHub Actions to push new versions to PyPI. This is already done by our own `python-timezones`: https://github.com/Doist/python-timezones/blob/main/.github/workflows/publish.yml

### PR DESCRIPTION
**Pull request by lukemerrett**
Fixes #24

Automating the release of the SQS Workers package whenever there is a new version tag pushed to remote.

* Updated dependency management and packaging to use Poetry
* Updated readme with instructions for using Poetry & bumping the version
* Added GitHub Action using [this as a reference](https://github.com/Doist/python-timezones/blob/main/.github/workflows/publish.yml)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -512,18 +516,20 @@ MemorySession should be even faster, but has all the limitations documented abov
...
 
 ## Releasing new versions
 
-- Bump version in `sqs_workers/__version__.py`
+- Bump version in `pyproject.toml`
 - Update the CHANGELOG
 - Commit the changes with a commit message "Version X.X.X"
-- Push the changes to GitHub and PyPI with a single command `make upload`
+- Tag the current commit with `vX.X.X`
+- Push the changes to GitHub with `git push origin vX.X.X`
```
Comment (Schnouki) [inline README.md:534]: More like: "create a release on GitHub named vX.X.X". It's a bit nicer, IMO.

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -19,13 +19,22 @@ jobs:
...
+
       - name: Set up Python
         uses: actions/setup-python@v2
         with:
           python-version: ${{ matrix.python-version }}
+
+      - name: Install poetry
+        run: |
+          python -m pip install poetry
+          poetry install
```
Comment (Schnouki) [inline .github/workflows/tests.yml:34]: This doesn't install `poetry`, but the project dependencies. (i.e., duplicate of what's just below.)

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -19,13 +19,22 @@ jobs:
       - name: Start Docker container with localstack
         if: matrix.session == 'localstack'
         run:  docker run -d -p 4566:9324 --rm softwaremill/elasticmq-native
+
       - name: Checkout Repo
         uses: actions/checkout@v3
+
       - name: Set up Python
         uses: actions/setup-python@v2
```
Comment (Schnouki) [inline .github/workflows/tests.yml:27]: Very minor nit: the "publish" workflow uses `setup-python@v4`, for the sake of consistency we should probably use the same here.

**Review by Schnouki — APPROVED** [review]
Very nice, thanks a lot Luke! There are a few minor details that could be improved, but nothing important. Feel free to merge and release once they are fixed 🚀

File: README.md (hunk 1 above)
Comment (lukemerrett) [reply README.md:534]: Ah ok makes sense 👍

**Pull request merged**